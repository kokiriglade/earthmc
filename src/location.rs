//! # Location
//!
//! Defines the [`LocationInfo`] struct.
use serde::{Deserialize, Serialize};

use crate::named_id::NamedIdOpt;

#[derive(Serialize, Deserialize, Debug, PartialEq)]
#[serde(rename_all = "camelCase")]
pub struct LocationInfo {
    pub location: PositionXZ,
    pub is_wilderness: bool,
    pub town: NamedIdOpt,
    pub nation: NamedIdOpt,
}

#[derive(Serialize, Deserialize, Debug, PartialEq)]
#[serde(rename_all = "camelCase")]
pub struct PositionXZ {
    pub x: f32,
    pub z: f32,
}
