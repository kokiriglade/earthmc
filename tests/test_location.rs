use earthmc::location::LocationInfo;

#[test]
fn test_location() {
    let raw_json = include_str!("inputs/location.json");
    let parsed: Vec<LocationInfo> = serde_json::from_str(raw_json).unwrap();
    insta::assert_debug_snapshot!(parsed);
}
