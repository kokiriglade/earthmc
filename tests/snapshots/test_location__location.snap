---
source: tests/test_location.rs
expression: parsed
---
[
    LocationInfo {
        location: PositionXZ {
            x: 0.0,
            z: 0.0,
        },
        is_wilderness: false,
        town: NamedIdGeneric {
            name: Some(
                "Sanctuary",
            ),
            uuid: Some(
                8dd45769-a6da-4035-a540-bb5d4a7184b8,
            ),
        },
        nation: NamedIdGeneric {
            name: None,
            uuid: None,
        },
    },
    LocationInfo {
        location: PositionXZ {
            x: 100.0,
            z: 100.0,
        },
        is_wilderness: true,
        town: NamedIdGeneric {
            name: None,
            uuid: None,
        },
        nation: NamedIdGeneric {
            name: None,
            uuid: None,
        },
    },
]
